src/getCypressElementCoordinates.ts

```diff
@@ -113,3 +103,59 @@ function scrollIntoView(htmlElement: HTMLElement, scrollBehavior: ScrollBehavior
 
     htmlElement.scrollIntoView({ block });
 }
+
+/**
+ * Returns the coordinates and size of a given Element, relative to the Cypress app <iframe>.
+ * Accounts for any scaling on the iframes.
+ */
+function getElementPositionXY(htmlElement: HTMLElement) {
+  const {
+    x: elementX,
+    y: elementY,
+    width,
+    height,
+  }= htmlElement.getBoundingClientRect();
+
+  const iframes = getContainingFrames(htmlElement) as HTMLElement[];
+
+  let frameScale = 1;
+  let finalX = 0;
+  let finalY = 0;
+
+  for (const frame of iframes) {
+    const { x, y, width } = frame.getBoundingClientRect();
+
+    finalX += x * frameScale;
+    finalY += y * frameScale;
+
+    frameScale = frameScale * (width / frame.offsetWidth);
+  }
+
+  finalX += elementX * frameScale;
+  finalY += elementY * frameScale;
+
+  return {
+    x: finalX,
+    y: finalY,
+    width: width * frameScale,
+    height: height * frameScale,
+  };
+}
+
+function getContainingFrames(element: HTMLElement) {
+  const iframes = [];
+
+  let currentWindow: Window | null = element.ownerDocument.defaultView;
+  while (currentWindow && currentWindow.frameElement && currentWindow !== window) {
```

Comment:
Just a note regarding `.frameElement` - it should return null for a cross-origin frame which could be a problem when running with `--disable-web-security` but I've checked that and it seems that this flag makes `.frameElement` return the correct element from the parent.